### PR TITLE
Orbit Menu: Add Auto-Observe mode! Refresh button! Alive/Ghost Counters! No more [ghost] tag!

### PR DESCRIPTION
**Pull request by MrDoomBringer**
## About The Pull Request
adds an auto-observe mode, something I've been wanting to do for a long time! Makes it so anything you orbit you observe as well. A button in the orbit menu toggles this on or off.

Also, adds a refresh button to the orbit menu. Not only is this performant - because we have a recursive orbitcounter now - but it also prevents you from accidently clicking on new mobs as the menu updates.

Also does some minor code improvement, ie. changing a couple 1s and 0s to TRUEs and FALSEs.
Finally, this fixes some bugs that mothblocks pointed out to me, shoutout to mothblocks

## Why It's Good For The Game
orbit menu....better....

## Changelog
:cl:
add: The orbit menu now has an Auto-Observe button! No more sifting through the lame observe menu to snoop in people's backpacks! Also, you can refresh the orbit menu now.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import { createSearch } from 'common/string';
 import { resolveAsset } from '../assets';
 import { useBackend, useLocalState } from '../backend';
-import { Box, Button, Flex, Icon, Input, Section } from '../components';
+import { Box, Button, Flex, Icon, Input, Section, Divider } from '../components';
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:4]:
```suggestion
import { Box, Button, Divider, Flex, Icon, Input, Section } from '../components';
```

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 2]
```diff
@@ -86,6 +85,7 @@ export const Orbit = (props, context) => {
     ghosts,
     misc,
     npcs,
+    auto_observe,
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:88]: Alphabetical order

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 3]
```diff
@@ -161,7 +181,7 @@ export const Orbit = (props, context) => {
           </Section>
         )}
 
-        <Section title="Alive">
+        <Section title={"Alive - (" + alive.length + ")"}>
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:184]:
```suggestion
        <Section title={`Alive - (${alive.length})`}>
```

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 4]
```diff
@@ -173,7 +193,7 @@ export const Orbit = (props, context) => {
             ))}
         </Section>
 
-        <Section title="Ghosts">
+        <Section title={"Ghosts - (" + ghosts.length + ")"}>
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:196]:
```suggestion
        <Section title={`Ghosts - (${ghosts.length})`}>
```

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 5]
```diff
@@ -140,6 +140,26 @@ export const Orbit = (props, context) => {
...
+            </Flex.Item>
+            <Flex.Item>
+              <Button
+                inline
+                color="transparent"
+                tooltip="Toggle Auto-Observe"
+                tooltipPosition="bottom-left"
+                selected={auto_observe}
+                icon={auto_observe === 1 ? "toggle-on" : "toggle-off"}
+                onClick={() => act('toggle_observe')} />
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:154]: It doesn't matter, but I just want the file to stay consistent.

```suggestion
                onClick={() => act("toggle_observe")} />
```

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 6]
```diff
@@ -140,6 +140,26 @@ export const Orbit = (props, context) => {
...
+              <Button
+                inline
+                color="transparent"
+                tooltip="Refresh"
+                tooltipPosition="bottom-left"
+                icon="sync-alt"
+                onClick={() => act('refresh')} />
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:161]:
```suggestion
                onClick={() => act("refresh")} />
```

File: tgui/packages/tgui/interfaces/Orbit.js [hunk 7]
```diff
@@ -140,6 +140,26 @@ export const Orbit = (props, context) => {
...
+              <Divider vertical />
+            </Flex.Item>
+            <Flex.Item>
+              <Button
+                inline
+                color="transparent"
+                tooltip="Toggle Auto-Observe"
+                tooltipPosition="bottom-left"
+                selected={auto_observe}
+                icon={auto_observe === 1 ? "toggle-on" : "toggle-off"}
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/Orbit.js:153]:
```suggestion
                icon={auto_observe ? "toggle-on" : "toggle-off"}
```

I like to pretend that BYOND has real booleans.

**Comment by ATH1909** [comment]
By the way, how DO you look in someone's backpack while you're observing them? I haven't been able to figure out how to do that yet...

**Comment by Jared-Fogle** [comment]
@ATH1909 Alt click.

**Comment by MrDoomBringer** [comment]
@Jared-Fogle Clicking normally works as well

**Review by Timberpoes — CHANGES_REQUESTED** [review]
```
Error: our build differs from the build committed into git.

Please rebuild tgui.
```

Please rebase and rebuild tgui before this can be merged.

**Comment by MrDoomBringer** [comment]
rebased and rebuilded tgui before this is merged

**Comment by stylemistake** [comment]
Rebuild and ping me to merge.

**Pull request merged**